### PR TITLE
V1.4 update dg

### PR DESCRIPTION
**Pull request by kaiyichen**
#215 - added manual testing for search and view
- updated search DG

https://kaiyichen.github.io/tp/DeveloperGuide.html

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -930,15 +930,73 @@ testers are expected to do more *exploratory* testing.
...
+       Expected: No applicant is displayed in the list. This is because search attribute has to be an exact word (can be split by spacing).
+
+    7. Test case: `search j/Bob`<br>
+       Expected: No applicant is displayed in the list. This is because the search attribute has to match with the correct prefix.
+
+    8. Test case: `search n/Tan j/Engineer`<br>
+       Expected: Both applicants are displayed in the list but `Bob Tan` is shown on the top of the list. This is because applicant with the most matching attributes will be displayed first.
+
+    9. Test case: `search n/Amy j/Engineer`<br>
+       Expected: Both applicants are displayed in the list and the order of applicant shown is based on the original list. In this case, `Bob Tan` is shown on top of `Amy Tan`.
```
Comment (goel-a) [inline docs/DeveloperGuide.md:971]:
```suggestion
       Expected: Both applicants are displayed in the list and the order of applicant shown is based on the original list. In this case, `Bob Tan` is shown above `Amy Tan`.
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -994,6 +994,64 @@ testers are expected to do more *exploratory* testing.
...
+    8. Test case: `search n/Tan j/Engineer`<br>
+       Expected: Both applicants are displayed in the list but `Bob Tan` is shown on the top of the list. This is because applicant with the most matching attributes will be displayed first.
+
+    9. Test case: `search n/Amy j/Engineer`<br>
+       Expected: Both applicants are displayed in the list and the order of applicant shown is based on the original list. In this case, `Bob Tan` is shown above `Amy Tan`.
+
+    10. Other incorrect search commands to try: `search`, `Search`, `search Bob`, `search w/Bob`, `search n/`<br>
+        Expected: No applicant is displayed. Error details shown in the status message.
+
+### View on an applicant
```
Comment (ryantanlien) [inline docs/DeveloperGuide.md:1031]: View ~~on~~ an applicant

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -994,6 +994,64 @@ testers are expected to do more *exploratory* testing.
...
+
+1. View on an applicant while all applicants are being shown in the *GUI*.
```
Comment (ryantanlien) [inline docs/DeveloperGuide.md:1033]: 1. View ~~on~~ an applicant while all applicants are being shown in the *GUI*.

**Review by ryantanlien — APPROVED** [review]
Minor grammatical error other than that LGTM!

**Pull request merged**